deploy/crds/push_v1alpha1_pushapplication_crd.yaml

```diff
@@ -16,35 +16,46 @@ spec:
     openAPIV3Schema:
       properties:
         apiVersion:
-          description: 'APIVersion defines the versioned schema of
-            this representation of an object. Servers should convert
-            recognized schemas to the latest internal value, and may
-            reject unrecognized values. More info:
-            https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
           type: string
         kind:
-          description: 'Kind is a string value representing the REST
-            resource this object represents. Servers may infer this
-            from the endpoint the client submits requests to. Cannot
-            be updated. In CamelCase. More info:
-            https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
           type: string
         metadata:
           type: object
         spec:
           properties:
             description:
-              description: 'Description is a description of the app to
-                be displayed in the UnifiedPush Server admin UI'
+              description: Description is a description of the app to be displayed
+                in the UnifiedPush Server admin UI
               type: string
+          required:
+          - description
+          type: object
         status:
           properties:
-            pushApplicationId:
-              type: string
             masterSecret:
+              description: MasterSecret is a master password, used for sending message
+                to this PushApplication, or it's Variant(s)
+              type: string
+            pushApplicationId:
+              description: PushApplicationId is an identifer used to register Variants
+                with this PushApplication
               type: string
             variants:
+              description: Variants is a slice of variant (AndroidVariant or IOSVariant,
```

Comment:
In this CRD case, is "slice" the best word, or would it be better as "array" or "list"?